src/opamp.rs

```diff
@@ -172,161 +238,213 @@ macro_rules! opamps {
                     }
                 }
 
-                /// States for opampX.
-                pub mod $opamp {
-                    use core::{borrow::Borrow, marker::PhantomData};
-
-                    #[allow(unused_imports)]
-                    use crate::gpio::gpioa::*;
+                impl $opamp {
+                    /// Reset the opamp. Used by internal implementations.
+                    unsafe fn _reset() {
+                        (*crate::stm32::OPAMP::ptr()).[<$opampreg _csr>].reset()
+                    }
 
-                    #[allow(unused_imports)]
-                    use crate::gpio::gpiob::*;
+                    /// Disable the output (connect the opamp to the internal ADC
+                    /// channel). Used by internal implementations.
+                    unsafe fn _disable_output() {
+                        (*crate::stm32::OPAMP::ptr()).[<$opampreg _csr>].write(|w|
```

Comment:
Is this and `_enable_output` supposed to overwrite(not modify) the entire csr register?

Reply:
No, good catch, fixed here: 13d4631

I made a small change to consolidate some of the code too. It results in an additional register write when constructing the opamps, but that seems like a negligible penalty.